app/pages/UcsfClientEligibilityPage/UcsfClientEligibilityPage.tsx

```diff
@@ -1,56 +1,30 @@
 import React, { useState } from 'react';
 import { useHistory } from 'react-router-dom';
 
-import { Checkbox } from 'components/ui/inline/Checkbox';
-import { Button } from 'components/ui/inline/Button';
-import { Section } from 'components/ucsf/Section';
-import { Layout } from 'components/ucsf/Layout';
+import { Checkbox } from 'components/ui/inline/Checkbox/Checkbox';
+import { Button } from 'components/ui/inline/Button/Button';
+import { Section } from 'components/ucsf/Section/Section';
+import { Layout } from 'components/ucsf/Layout/Layout';
 
+import { eligibilityData } from './ucsfEligibilities';
 import styles from './UcsfClientEligibilityPage.module.scss';
 
-// import { useEligibilitiesForCategory } from '../../hooks/APIHooks';
-
-interface clientEligibility {
-  checked: boolean;
-  name: string;
-}
-
-interface clientEligibilityGroup {
-  label: string;
-  eligibilities: clientEligibility[];
-}
-
-// Todo: This is dummy data for development. It will be replaced by data returned by API request(s)
-const clientEligibilityGroupData: clientEligibilityGroup[] = [
-  {
-    label: 'Age and Dependents',
-    eligibilities: [
-      { checked: false, name: 'See all' },
-      { checked: false, name: 'Under 18' },
-      { checked: false, name: 'I am a single adult and need shelter' },
-    ],
-  },
-  {
-    label: 'Gender Identity',
-    eligibilities: [
-      { checked: false, name: 'See all' },
-      { checked: false, name: 'Woman' },
-      { checked: false, name: 'Man' },
-      { checked: false, name: 'Transgender' },
-    ],
-  },
-  {
-    label: 'Health Related',
-    eligibilities: [
-      { checked: false, name: 'See all' },
-      { checked: false, name: 'HIV' },
-      { checked: false, name: 'Dual Diagnosis' },
-    ],
-  },
-];
-
-const ClientEligibilities = () => {
-  const [eligibilityGroupList, setEligibilityGroupList] = useState(clientEligibilityGroupData);
+const ClientEligibilities = ({ rawEligibilityData, resourceSlug }: {
+  rawEligibilityData: any;
+  resourceSlug: string;
+}) => {
+  interface clientEligibility {
+    checked: boolean;
+    name: string;
+  }
+
+  interface clientEligibilityGroup {
+    label: string;
+    eligibilities: clientEligibility[];
+  }
+
+  const resourceEligibilityGroup = rawEligibilityData[resourceSlug];
+  const [eligibilityGroupList, setEligibilityGroupList] = useState(resourceEligibilityGroup);
```

Comment:
What you're doing here is a little bit dangerous, since it has a risk of caching the state longer than you want. Specifically, if `resourceSlug` changes, the state won't be updated to contain the correct set of eligibilities. Out of sheer luck, as currently structured, the app can't run into that case here, since currently the `ClientEligibilities` component is guaranteed to be unmounted before being remounted with a new prop value. However, I think we should still fix this, since it could lead to some very weird bugs in the future if we change how this component is instantiated by its parent.

What you'll want to do is add a `useEffect()` call that triggers when the `resourceSlug` prop changes and that resets this eligibilityGroupList state. See https://bobbyhadz.com/blog/react-update-state-when-props-change for a simple example of this.

Reply:
Awesome thanks for explaining this subtle issue and providing the link. I have changed it per your suggestion.